### ISSUE TITLE
generating spectra with 'point' region type results in negative x_wid value

### ISSUE THREAD

**Issue by TobyWallage** [DavidT3/XGA#1028]
Using
-  current XGA version from git repository;
- SAS version 21.0.0 which could be the cause of this error in which case this may be a feature request for compatibility with future SAS versions

For context, I was following the tutorial notebook for spectroscopy.

``` python
demo_src = PointSource(182.2416, 45.67667, 0.9254, name='PG1206+459')

demo_src = evselect_spectrum(demo_src, 'point')

demo_src = power_law(demo_src, 'point', redshifted=True)

for sp in demo_src.get_spectra('point'):
    sp.view()
```

This code was failing to plot the errorbars when calling the view methods due to `x_err` being negative.
Upon investigating, the `power_law` fitting was also silently failing.
Manually printing the `x_wid` array which is used as the error in the view method showed the culprit to be the last value in the array for all the spectra;
This is the printed x_wid array for 0033540601PN Spectrum, though MOS1 and MOS2 were similarly affected
```python
array([ 2.62950361e-03,  2.62750685e-03,  5.24950027e-03,  2.62250006e-03,
        5.23899496e-03,  2.61649489e-03,  1.04490072e-02,  7.81700015e-03,
        7.79950619e-03,  5.19049168e-03,  7.77098536e-03,  5.17101586e-03,
        2.58250535e-03,  2.58047879e-03,  5.15550375e-03,  7.71850348e-03,
        5.13601303e-03,  1.28054768e-02,  5.10850549e-03,  2.55149603e-03,
        5.09750843e-03,  7.63151050e-03,  5.07798791e-03,  5.07101417e-03,
        7.59199262e-03,  1.00975037e-02,  5.03799319e-03,  5.03152609e-03,
        1.00409687e-02,  1.00150108e-02,  4.99749184e-03,  7.48452544e-03,
        9.95746255e-03,  7.45251775e-03,  4.96149063e-03,  4.95550036e-03,
        7.42301345e-03,  1.23459995e-02,  4.92948294e-03,  4.92501259e-03,
        9.83601809e-03,  4.91148233e-03,  7.35950470e-03,  7.35050440e-03,
        9.78749990e-03,  1.22149885e-02,  1.21960044e-02,  9.74398851e-03,
        4.86850739e-03,  9.72950459e-03,  1.45825148e-02,  4.85497713e-03,
        1.21349692e-02,  4.85002995e-03,  1.45500302e-02,  4.84997034e-03,
        1.45400167e-02,  1.21150017e-02,  1.45399570e-02,  1.45400763e-02,
        4.84997034e-03,  4.85002995e-03,  7.26991892e-03,  1.21300220e-02,
        9.71001387e-03,  1.21400356e-02,  9.71496105e-03,  9.72497463e-03,
        9.73004103e-03,  1.46099925e-02,  7.30496645e-03,  1.21900439e-02,
        7.31998682e-03,  1.95399523e-02,  1.46750808e-02,  1.22449994e-02,
        1.22550130e-02,  1.71749592e-02,  7.36999512e-03,  1.22900009e-02,
        1.47650242e-02,  2.21749544e-02,  9.86999273e-03,  2.22250223e-02,
        1.97849870e-02,  2.22899914e-02,  1.98299885e-02,  3.22750211e-02,
        2.48550177e-02,  2.48800516e-02,  9.95993614e-03,  3.23799849e-02,
        3.24100256e-02,  2.74449587e-02,  4.99200821e-02,  2.49749422e-02,
        6.24550581e-02,  7.24749565e-02,  2.24950314e-02,  7.24899769e-02,
        2.75000334e-02,  6.24949932e-02,  3.24999094e-02,  2.25000381e-02,
        5.50000668e-02,  8.99999142e-02,  5.25000095e-02,  3.50000858e-02,
        6.74999952e-02,  5.49999475e-02,  7.99999237e-02,  5.50000668e-02,
        7.99999237e-02,  7.24999905e-02,  1.67500019e-01,  1.37500048e-01,
        6.99999332e-02,  1.05000019e-01,  1.47500038e-01,  9.00001526e-02,
        4.49998379e-02,  1.44999981e-01,  1.74999952e-01,  5.75001240e-02,
        3.25000048e-01,  1.77500010e-01,  1.54999971e-01, -6.13508204e+00])
```

I also manually modified the view method to use the absolute value of the x_err in order to force the spectrum plot;

Forced Spectrum plot;
![43241f93-ce2b-4a2a-a9c1-265b34eac568](https://github.com/DavidT3/XGA/assets/106962588/e3205930-9ce7-4d47-bc10-f8968fbdb64d)

I suspect this may be due to my SAS version 21.0.0 being incompatible however the cluster sample used in the Spectroscopy tutorial notebook ***did not*** fail so it may be a problem in selecting 'point' regions for spectrum.

**Comment by TobyWallage** [comment]
So originally I thought XSpec was failing due to the spectra generated by SAS, turns out it was actually failing due to the issue fixed in #942 (*which was why I switched from the release version to the current github version*) and me not properly removing my `xga_output` files and xga just reusing the old XSpec scripts.

This caused the model fits to fail and the `Spectrum.view()` method to fall-back to `show_model_fits=False` behaviour. Just nuking every `xga_ouput` file I could find on my system was enough to fix the code I was using to produce the expected output. 

However, I still think there may be a problem with the `Spectrum.view()` method as manually disabling the model fits with 
``` python
sp.view(show_model_fits=False)
```
Still causes a negative xerr value for the errorbar plot.
I still wonder if this may be due to my SAS version but need someone else to test this...

**Comment by DavidT3** [comment]
Thanks for writing this out so nicely Toby, and I'm sorry for the slow response, I've been very busy recently. I'm also coming across what sounds like the same problem in one of my spectral plots:

ValueError                                Traceback (most recent call last)
Cell In[4], line 5
      3 for sp in src.get_spectra(src.r500):
      4     print(sp.obs_id, sp.instrument)
----> 5     sp.view()
      6 print('\n\n\n')

File /mnt/ufs18/home-218/turne540/code/XGA/xga/products/spec.py:1662, in Spectrum.view(self, figsize, lo_lim, hi_lim, back_sub, energy, src_colour, bck_colour, grouped, xscale, yscale, fontsize, show_model_fits, save_path)
   1658 if not show_model_fits:
   1659     # Plotting the data, accounting for the different combinations of x-axis and y-axis
   1660     if back_sub:
   1661         # If we're going for background subtracted data, then that is all we plot
-> 1662         plt.errorbar(x_dat, src_sub_bck_rate.value[:, 0] / per_x, xerr=x_wid,
   1663                      yerr=src_sub_bck_rate.value[:, 1] / per_x, fmt="+", color=src_colour,
   1664                      label="Background subtracted source data", zorder=1)
   1665     else:
   1666         # But if we're not wanting background subtracted, we need to plot the source and background spectra
   1667         plt.errorbar(x_dat, src_rate.value[:, 0] / per_x, xerr=x_wid, yerr=src_rate.value[:, 1] / per_x, fmt="+",
   1668                      color=src_colour, label="Source data", zorder=1)

File ~/software/anaconda3/envs/xga_dev/lib/python3.9/site-packages/matplotlib/pyplot.py:2564, in errorbar(x, y, yerr, xerr, fmt, ecolor, elinewidth, capsize, barsabove, lolims, uplims, xlolims, xuplims, errorevery, capthick, data, **kwargs)
   2558 @_copy_docstring_and_deprecators(Axes.errorbar)
   2559 def errorbar(
   2560         x, y, yerr=None, xerr=None, fmt='', ecolor=None,
   2561         elinewidth=None, capsize=None, barsabove=False, lolims=False,
   2562         uplims=False, xlolims=False, xuplims=False, errorevery=1,
   2563         capthick=None, *, data=None, **kwargs):
-> 2564     return gca().errorbar(
   2565         x, y, yerr=yerr, xerr=xerr, fmt=fmt, ecolor=ecolor,
   2566         elinewidth=elinewidth, capsize=capsize, barsabove=barsabove,
   2567         lolims=lolims, uplims=uplims, xlolims=xlolims,
   2568         xuplims=xuplims, errorevery=errorevery, capthick=capthick,
   2569         **({"data": data} if data is not None else {}), **kwargs)

File ~/software/anaconda3/envs/xga_dev/lib/python3.9/site-packages/matplotlib/__init__.py:1442, in _preprocess_data.<locals>.inner(ax, data, *args, **kwargs)
   1439 @functools.wraps(func)
   1440 def inner(ax, *args, data=None, **kwargs):
   1441     if data is None:
-> 1442         return func(ax, *map(sanitize_sequence, args), **kwargs)
   1444     bound = new_sig.bind(ax, *args, **kwargs)
   1445     auto_label = (bound.arguments.get(label_namer)
   1446                   or bound.kwargs.get(label_namer))

File ~/software/anaconda3/envs/xga_dev/lib/python3.9/site-packages/matplotlib/axes/_axes.py:3642, in Axes.errorbar(self, x, y, yerr, xerr, fmt, ecolor, elinewidth, capsize, barsabove, lolims, uplims, xlolims, xuplims, errorevery, capthick, **kwargs)
   3639 res = np.zeros(err.shape, dtype=bool)  # Default in case of nan
   3640 if np.any(np.less(err, -err, out=res, where=(err == err))):
   3641     # like err<0, but also works for timedelta and nan.
-> 3642     raise ValueError(
   3643         f"'{dep_axis}err' must not contain negative values")
   3644 # This is like
   3645 #     elow, ehigh = np.broadcast_to(...)
   3646 #     return dep - elow * ~lolims, dep + ehigh * ~uplims
   3647 # except that broadcast_to would strip units.
   3648 low, high = dep + np.row_stack([-(1 - lolims), 1 - uplims]) * err

ValueError: 'xerr' must not contain negative values
